src/main/java/laddergame/domain/Line.java

```diff
@@ -0,0 +1,35 @@
+package laddergame.domain;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static laddergame.messsages.ExceptionMessages.LINE_POINTS_BOTH_TRUE_EXCEPTION;
+import static laddergame.messsages.ExceptionMessages.LINE_POINTS_EMPTY_EXCEPTION;
+
+public class Line {
+    private final List<Boolean> points;
+
+    public Line(final List<Boolean> points) {
+        validateNotEmpty(points);
+        validateNotBothTrue(points);
+        this.points = points;
+    }
+
+    private void validateNotBothTrue(final List<Boolean> points) {
+        boolean isBothTrue = IntStream.range(0, points.size() - 1)
+                .anyMatch(pointIndex -> points.get(pointIndex) && points.get(pointIndex + 1));
+        if (isBothTrue) {
+            throw new IllegalArgumentException(LINE_POINTS_BOTH_TRUE_EXCEPTION.getMessage());
+        }
+    }
```

Comment:
```suggestion
  private void validatePoints(final List<Boolean> points) {
    if (hasConsecutivePositivePoints(points)) {
      throw new IllegalArgumentException(LINE_POINTS_BOTH_TRUE_EXCEPTION.getMessage());
    }
  }

  private static boolean hasConsecutivePositivePoints(List<Boolean> points) {
    return IntStream.range(0, points.size() - 1)
        .anyMatch(pointIndex -> {
          var basePoint = points.get(pointIndex);
          var nextPoint = points.get(pointIndex + 1);
          return basePoint && nextPoint;
            }
        );
  }
```
는 어떨까요? 

validate의 구현까지 바로 이해하지 않아도 될 것 같아요 :) 

- 기존
Line을 만들기 위해서 -> "validateNotBothTrue"를 하네, -> validateNotBothTrue는, Boolean 리스트의 연속된 2개의 값을 비교했을 때, true인 값이 하나라도 존재를 한다면, 예외를 뱉네.

- 수정
Line을 만들기 위해서 -> validatePoint를 하네. -> 연속된 양수를 가진다면 예외를 뱉네 -> (이후에) 연속된 양수를 가지는 구현은 어떻게 되있을까 궁금하네? 

추상화 정도를 적절히 설정하면, 읽기 쉬운 코드를 작성할 수 잇어요 :) (구현은 어떻게 작성해도 읽기 어렵답니다 )

Reply:
헉.. 윗 내용들로 수정하고 나서 봤네요. suggestion까지 작성해주셔서 감사합니다. 근데 조언 주신 내용들로 먼저 리팩토링 했을 때 비슷한 형태가 된 것같아 신기합니다 ㅎㅎ

Reply:
잘 리팩터링 해주신다면, 점점 더 완성되어가는게 보일꺼에요 :)